lib/broccoli-cldr.js

```diff
@@ -0,0 +1,79 @@
+/* jshint node: true */
+
+/**
+* Copyright 2015, Yahoo! Inc.
+* Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
+*/
+
+'use strict';
+
+var cldr          = require('formatjs-extract-cldr-data');
+var CachingWriter = require('broccoli-caching-writer');
+var objectAssign  = require('object.assign');
+var mkdirp        = require('mkdirp');
+var path          = require('path');
+var fs            = require('fs');
+var walkSync      = require('walk-sync');
+
+function LocaleWriter (inputTrees, outputPath, options) {
+    if (!(this instanceof LocaleWriter)) {
+        return new LocaleWriter(inputTrees, outputPath, options);
+    }
+
+    if (!Array.isArray(inputTrees)) {
+        inputTrees = [inputTrees];
+    }
+
+    CachingWriter.call(this, inputTrees, options);
+
+    this.options = objectAssign({
+        locales:        undefined,
+        pluralRules:    false,
+        relativeFields: false,
+        prelude:        '',
+        wrapEntry:      undefined // todo: behaves slightly different, rename to transform?
+    }, options);
+
+    this.outputPath = outputPath;
+}
+
+LocaleWriter.prototype = Object.create(CachingWriter.prototype);
+LocaleWriter.prototype.constructor = LocaleWriter;
+
+LocaleWriter.prototype.updateCache = function (includePaths, destDir) {
+    var destPath    = path.join(destDir, this.outputPath);
+    var options     = this.options;
+    var wrapEntry   = options.wrapEntry;
+    var localesData = cldr(options);
+
+    mkdirp.sync(destPath);
+
+    Object.keys(localesData).forEach(function (localeKey) {
+        var localeData = localesData[localeKey];
+        var data       = localeData;
+
+        if (typeof wrapEntry === 'function') {
+            data = wrapEntry(localeData.locale, {
+                parentLocale: localeData.parentLocale,
+                pluralFn:     localeData.pluralRuleFunction,
+                fields:       localeData.fields
+            });
+        }
+
+        var outFile = path.join(destPath, localeKey.toLowerCase() + '.js');
```

Comment:
> I don't think you want to do this because there will be missing information when a locale like en-US delegates to its parent en for all the real data.

If these files were to be written to the file system and kept as static files, this would be the case.  What this is doing ultimately is writing to the "application tree" by broccoli, which get bundled into app.js all together.  So if my app locale is 'en-us', it will write 'en-us' and 'en' with 'en-us' not having any data other than `parentLocale` set to `en`.

When the application is initialized I call addLocaleData on each of the CLDR objects written to the app tree.  What you mention is in fact an issue though when pulling out this broccoli plugin for others to have more control over this extraction process.  One use-case I've heard is the need to write them to the asset tree, so that they are not bundled at all .. instead individually stored.  That's where your proposal of "languages" comes in and will definitely solve this.